integration/pushed_authorize_code_grant_test.go

```diff
@@ -0,0 +1,202 @@
+
+	var state string
+	for k, c := range []struct {
+		description    string
+		setup          func()
+		check          func(t *testing.T, r *http.Response)
+		params         map[string]string
+		authStatusCode int
+		parStatusCode  int
+	}{
+		{
+			description: "should fail because of audience",
+			params:      map[string]string{"audience": "https://www.ory.sh/not-api"},
+			setup: func() {
+				oauthClient = newOAuth2Client(ts)
+				state = "12345678901234567890"
+			},
+			parStatusCode:  http.StatusBadRequest,
+			authStatusCode: http.StatusNotAcceptable,
+		},
+		{
+			description: "should fail because of scope",
+			params:      nil,
+			setup: func() {
+				oauthClient = newOAuth2Client(ts)
+				oauthClient.Scopes = []string{"not-exist"}
+				state = "12345678901234567890"
+			},
+			parStatusCode:  http.StatusBadRequest,
+			authStatusCode: http.StatusNotAcceptable,
+		},
+		{
+			description: "should pass with proper audience",
+			params:      map[string]string{"audience": "https://www.ory.sh/api"},
+			setup: func() {
+				oauthClient = newOAuth2Client(ts)
+				state = "12345678901234567890"
+			},
+			check: func(t *testing.T, r *http.Response) {
+				var b fosite.AccessRequest
+				b.Client = new(fosite.DefaultClient)
+				b.Session = new(defaultSession)
+				require.NoError(t, json.NewDecoder(r.Body).Decode(&b))
+				assert.EqualValues(t, fosite.Arguments{"https://www.ory.sh/api"}, b.RequestedAudience)
+				assert.EqualValues(t, fosite.Arguments{"https://www.ory.sh/api"}, b.GrantedAudience)
+				assert.EqualValues(t, "foo-sub", b.Session.(*defaultSession).Subject)
+			},
+			parStatusCode:  http.StatusCreated,
+			authStatusCode: http.StatusOK,
+		},
+		{
+			description: "should pass",
+			setup: func() {
+				oauthClient = newOAuth2Client(ts)
+				state = "12345678901234567890"
+			},
+			parStatusCode:  http.StatusCreated,
+			authStatusCode: http.StatusOK,
+		},
+	} {
+		t.Run(fmt.Sprintf("case=%d/description=%s", k, c.description), func(t *testing.T) {
+			c.setup()
+
+			// build request from the OAuth client
+			data := url.Values{}
+			data.Set("client_id", oauthClient.ClientID)
+			data.Set("client_secret", oauthClient.ClientSecret)
+			data.Set("response_type", "code")
+			data.Set("state", state)
+			data.Set("scope", strings.Join(oauthClient.Scopes, " "))
+			data.Set("redirect_uri", oauthClient.RedirectURL)
+			for k, v := range c.params {
+				data.Set(k, v)
+			}
+
+			req, err := http.NewRequest("POST", ts.URL+"/par", strings.NewReader(data.Encode()))
+			require.NoError(t, err)
+
+			req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+			resp, err := http.DefaultClient.Do(req)
+
+			require.NoError(t, err)
+
+			body, err := checkStatusAndGetBody(t, resp, c.parStatusCode)
+			require.NoError(t, err, "Unable to get body after PAR. Err=%v", err)
+
+			if resp.StatusCode != http.StatusCreated {
+				return
+			}
```

Comment:
Shouldn't this be a failure instead? Right now it will just silently pass the test

Reply:
The failure occurs before this call when we assert the HTTP status code.

Reply:
I see, we're explicitly testing for the response code before - make sense